hls-plugin-api/src/Ide/Types.hs

```diff
@@ -270,32 +300,82 @@ instance PluginMethod TextDocumentCompletion where
+    pluginResponsible uri pluginDesc && PluginId (formattingProvider conf) == pid
+    where
+      uri = msgParams ^. J.textDocument . J.uri
+      pid = pluginId pluginDesc
+  combineResponses _ _ _ _ x = sconcat x
+
 
 instance PluginMethod TextDocumentRangeFormatting where
-  pluginEnabled _ pid conf = (PluginId $ formattingProvider conf) == pid
+  pluginEnabled _ msgParams pluginDesc conf = pluginResponsible uri pluginDesc
+      && PluginId (formattingProvider conf) == pid
+    where
+      uri = msgParams ^. J.textDocument . J.uri
+      pid = pluginId pluginDesc
   combineResponses _ _ _ _ (x :| _) = x
 
 instance PluginMethod TextDocumentPrepareCallHierarchy where
-  pluginEnabled _ = pluginEnabledConfig plcCallHierarchyOn
+  pluginEnabled _ msgParams pluginDesc conf = pluginResponsible uri pluginDesc
+      && pluginEnabledConfig plcCallHierarchyOn pid conf
+    where
+      uri = msgParams ^. J.textDocument . J.uri
+      pid = pluginId pluginDesc
 
 instance PluginMethod TextDocumentSelectionRange where
-  pluginEnabled _ = pluginEnabledConfig plcSelectionRangeOn
+  pluginEnabled _ _ pluginDesc conf = pluginEnabledConfig plcSelectionRangeOn pid conf
+    where
+      pid = pluginId pluginDesc
   combineResponses _ _ _ _ (x :| _) = x
 
 instance PluginMethod CallHierarchyIncomingCalls where
-  pluginEnabled _ = pluginEnabledConfig plcCallHierarchyOn
+  pluginEnabled _ _ pluginDesc conf = pluginEnabledConfig plcCallHierarchyOn pid conf
+    where
+      pid = pluginId pluginDesc
 
 instance PluginMethod CallHierarchyOutgoingCalls where
-  pluginEnabled _ = pluginEnabledConfig plcCallHierarchyOn
+  pluginEnabled _ _ pluginDesc conf = pluginEnabledConfig plcCallHierarchyOn pid conf
+    where
+      pid = pluginId pluginDesc
 
 instance PluginMethod CustomMethod where
-  pluginEnabled _ _ _ = True
+  pluginEnabled _ _ _ _ = True
   combineResponses _ _ _ _ (x :| _) = x
 
 -- ---------------------------------------------------------------------
 
+class HasTracing (MessageParams m) => PluginNotificationMethod (m :: Method FromClient Notification)  where
+  pluginEnabled2 :: SMethod m -> MessageParams m -> PluginDescriptor c -> Config -> Bool
+
+  default pluginEnabled2 :: (HasTextDocument (MessageParams m) doc, HasUri doc Uri)
+                              => SMethod m -> MessageParams m -> PluginDescriptor c -> Config -> Bool
+  pluginEnabled2 _ params desc conf = pluginResponsible uri desc && plcGlobalOn (configForPlugin conf (pluginId desc))
+    where
+        uri = params ^. J.textDocument . J.uri
+
+instance PluginNotificationMethod TextDocumentDidOpen where
+
+instance PluginNotificationMethod TextDocumentDidChange where
+
+instance PluginNotificationMethod TextDocumentDidSave where
+
+instance PluginNotificationMethod TextDocumentDidClose where
+
+instance PluginNotificationMethod WorkspaceDidChangeWatchedFiles where
+  pluginEnabled2 _ _ desc conf = plcGlobalOn $ configForPlugin conf (pluginId desc)
```

Comment:
I'm a bit dubious about skipping some file notifications and not others. Why not filter the watched file notifications too?

Reply:
Yeah, I don't remember why it is this way...

Reply:
This also seems a bit risky, in that e.g. this means that HLS presumably won't track cabal files in its VFS, which... might be okay? Or might not be.

Reply:
We do want it to track it in its VFS. But why does this line affect the VFS?

Reply:
Um, maybe I'm getting confused. This will stop the `ghcide` file change notification handlers from picking it up, but I forgot that `lsp` installs its own handlers that maintain the VFS.